shaker/libs/github.py

```diff
@@ -363,17 +416,62 @@ def resolve_constraint_to_object(org_name, formula_name, constraint):
     shaker.libs.logger.Logger().debug("github::resolve_constraint_to_object: "
                                       "resolve_constraint_to_object(%s, %s, %s)"
                                       % (org_name, formula_name, constraint))
-    wanted_tag, tag_versions, tags_data = get_valid_tags(org_name, formula_name)
 
+    # do we have a constraint?
+    if constraint:
+        # is it a branch or a tag?
+        shaker.libs.logger.Logger().debug("github::resolve_constraint_to_object: "
+                                          "constraint is not empty '%s'"
+                                          % (constraint))
+        parsed_constraint = metadata.parse_constraint(constraint)
+        shaker.libs.logger.Logger().debug("github::resolve_constraint_to_object: "
+                                          "parsed_constraint '%s'"
+                                          % (str(parsed_constraint)))
+        # is it a branch (i.e. not a version)
+        if not parsed_constraint['version']:
+            branch_name = parsed_constraint['tag']
+            shaker.libs.logger.Logger().debug("github::resolve_constraint_to_object: "
+                                              "There is no version, assuming this is "
+                                              "a branch, name: '%s'"
+                                              % (branch_name))
+            branch_data = get_branch_data(org_name, formula_name, branch_name)
+            if branch_data:
+                obj = {
```

Comment:
When requesting branch details, Github returns a lengthy json which contains a lot of entries which are not used in this context. 
I decided to get the essential values, to keep the object similar to the one returned when the requirement is a version.